apitools/org.eclipse.pde.api.tools.ui/src/org/eclipse/pde/api/tools/ui/internal/actions/CollapseAllAction.java

```diff
@@ -24,7 +24,7 @@
  */
 public class CollapseAllAction extends Action {
 
-	private TreeViewer fViewer;
+	private final TreeViewer fViewer;
```

Comment:
does this compile? AFAIK, this produces `unitialized final field` error.

Reply:
It's assigned in the constructor, so yes it compiles without errors:

![image](https://github.com/eclipse-pde/eclipse.pde/assets/208716/cdf09f70-6316-4317-910b-57ace85e05d6)